code/modules/surgery/tools.dm

```diff
@@ -524,13 +524,13 @@
 	. = TRUE
 	switch(action)
 		if("add")
-			var/new_chem_name = params["name"]
-			var/chem_id = get_chem_id(new_chem_name)
-			if(!chem_id || !new_chem_name)
-				to_chat(usr, span_warning("No such known reagent exists!"))
-				return
-			if(!(chem_id in whitelist))
-				whitelist[chem_id] = new_chem_name
+			var/selected_reagent = tgui_input_list(usr, "Select Reagent To Filter", "WhiteList Reagent", GLOB.chemical_name_list)
+			if(selected_reagent)
+				var/chem_id = get_chem_id(selected_reagent)
+				if(chem_id && !(chem_id in whitelist))
+					whitelist[chem_id] = selected_reagent
+				else
+					to_chat(usr, span_warning("Invalid Reagent Or Already In List!"))
```

Comment:
CRASH here

Reply:
So u replace the code inside else condition with CRASH()? Sure but I don't think else condition will ever be executed since it will always return a valid name from the list. Maybe I remove the else condition all together?

Reply:
It is up to you--it never being executed is why you would want a CRASH there, so we get a stack trace when it happens, but it's not necessary and would only be if you find it useful (or if the invariant is something shakier)